### ISSUE TITLE
Which files are pulled?

### ISSUE THREAD

**Issue by jakobliebmann** [tillschwoerer/R-data-science-book#21]
Hi all,

I would like to start a discussion: which files make sense to "maintain" in the repo. Should the teams be limited to their specific rmd-file, or does it make sense to update "everything" (including the "generated" HTML files) with every pull request?
My first evaluation is as follows:
Advantage of the lean approach: less confusion during review
Advantage of the Full-Knit-Merge: the possibility to view the current HTML product online
I would be pleased about active participation.

Best, Jakob

**Comment by tillschwoerer** [comment]
Good point, I think I am slighly in favour of the lean approach. We could still build the most recent version of the book locally. But when we commit our changes, we would only commit the changes to the one Rmd file that we were working on, and not the many html files generated when the book is knitted. We would only commit the html files when the book is finished, or when a certain milestone is reached.

**Comment by tillschwoerer** [comment]
Yet another idea would be to commit the `.Rmd`  file, the corresponding `.md` file, and if plots are involved all generated/modified `.png` files. 

**Let me explain the context**

I just experimented (in a new branch `separate-sessions`) with a different `_bookdown.yml` configuration, by setting `new_session: yes`.  Now every chapter gets knitted in a separate R session, which has the advantage of removing dependencies between chapters. Now, changes in other peoples' chapters cannot have side effects on your chapter. Note that this approach has also a slight drawback that I could not yet resolve (described in issue #9 ).

The change in the config has the following other effect: When e.g. Carl changes `forcats.Rmd` and then presses knit or build the book, a corresponding `forcats.md` file is created/modified in the docs folder (which is an intermediate output required to generate `forcats.html`). Additionally, the plots generated by `forcats.Rmd` are stored as `*.png` files in `docs/forcats_files/figure-html`. Now, my suggestion would be to commit `forcats.Rmd`, `forcats.md` and the modified `*.png` files.

**What would be the advantage?** 

This allows your reviewer to visually inspect not only your code, but also your output (!) directly on GitHub. Open this [link to the forcats.md file](https://github.com/tillschwoerer/R-data-science-book/blob/separate-sessions/docs/forcats.md) to verify. Also you can [visually inspect changes in plots](https://help.github.com/en/github/managing-files-in-a-repository/rendering-and-diffing-images). This would make the review directly on GitHub easier.
